src/server.rs

```diff
@@ -474,6 +482,8 @@ impl Server {
         // Clear the buffer for next query.
         self.buffer.clear();
 
+        self.latest_successful_server_interaction_timestamp = SystemTime::now();
```

Comment:
Getting system time is relatively expensive (it's a syscall), so we should minimize those as much as possible. For example, pgbouncer caches the value of time for the duration of the loop.